Resources/config/phpcr.xml

```diff
@@ -61,15 +61,12 @@
         </service>
 
         <service id="doctrine_phpcr" class="%doctrine_phpcr.class%">
-            <argument>PHPCR</argument>
+            <argument type="service" id="service_container" />
             <argument>%doctrine_phpcr.sessions%</argument>
             <argument>%doctrine_phpcr.odm.document_managers%</argument>
             <argument>%doctrine_phpcr.default_session%</argument>
             <argument>%doctrine_phpcr.odm.default_document_manager%</argument>
             <argument>%doctrine_phpcr.proxy.class%</argument>
-            <call method="setContainer">
```

Comment:
indeed, dunno why we did it like this and not always inject the container in the constructor... thanks!

Reply:
This is needed to go on symfony 4.0 i think also needed for PSR-11

Reply:
As I wrote in #285, what about using a ServiceLocator (introduced in Symfony 3.3)?

Reply:
But how to test it? I can inject a defined set of services the locator could serve right?

Reply:
lets keep it simple here to keep working as it did before. service locator can be tried as a separate pull request once this is merged. ok?